src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndexForKey(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (size + 1 >= threshold) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> previousNode = table[index];
+            while (true) {
+                if (Objects.equals(previousNode.key, key)) {
+                    previousNode.value = value;
+                    return;
+                }
+                if (previousNode.next == null) {
+                    break;
+                } else {
+                    previousNode = previousNode.next;
+                }
+            }
+            previousNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = getIndexForKey(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                break;
+            }
+            node = node.next;
+        }
+        return node == null ? null : node.value;
```

Comment:
```suggestion
    while (node != null) {
            if (Objects.equals(node.key, key)) {
                return node.value;
            }
            node = node.next;
        }
        return null;
```

Reply:
It's not a mistake, it is just my preferred style code. I don't like several `return` in functions. But I'll ask about it on QnA

Reply:
IMHO I think `return`s are more readable, with them you do not need to read the other part of the logic in some cases and you understand right away that that's where the method exits the execution, which may be useful when you have a method on 300+ lines for example. Also `return`s and `break`s are not completely interchangeable, so you need to be mindful of that. Overall I would say that this is not that big of a deal, but most of the other community may prefer the `return` as well tho